codec-http/src/main/java/io/netty/handler/codec/http/cookie/ClientCookieEncoder.java

```diff
@@ -109,7 +109,7 @@ public int compare(Cookie c1, Cookie c2) {
             }
             // Rely on Java's sort stability to retain creation order in cases where
             // cookies have same path length.
-            return -1;
+            return 0;
```

Comment:
This might be problematic since ordering of same-length cookies will be inconsistent/non-deterministic. Cookies could also be "lost" if added to a `SortedMap`. Interestingly the current logic contradicts the javadoc above 
`breaking ties by sorting into increasing chronological order of creation time, as recommended by RFC 6265. order of creation time, as recommended by RFC 6265.` ... maybe that was the original intention but found to be not possible because creation time isn't stored in the Cookie objects.

I would suggest at minimum something like `return c1.name().compareTo(c2.name());` instead, but to be completely correct additional tiebreaks should be done using the path and domain fields' content too...

Reply:
(but thank you for finding the bug @gerdriesselmann, -1 is not correct either!)

Reply:
@njhill can you prepare a PR ?

Reply:
@gerdriesselmann, @normanmaurer apologies I think my suggestion was bad ... since this is a private comparator used in specific places for `Arrays#sort()` only, the concerns I have are not really applicable.

I.e. the original author's intended approach seems reasonable and this is the correct fix for that. My only remaining suggestion would be to amend the comment above to say that it's relying on `Arrays.sort`'s stability rather than "Java".

Reply:
ok lets do this...